client/galaxy/scripts/mvc/tool/tool-form-base.js

```diff
@@ -151,19 +94,9 @@ define( [ 'utils/utils', 'utils/deferred', 'mvc/ui/ui-misc', 'mvc/form/form-view
                             icon    : 'fa-cube',
                             onclick : function() {
                                 // here we update the tool version (some tools encode the version also in the id)
-                                var id = options.id.replace( options.version, this.version );
-                                var version = this.version;
-                                // queue model request
-                                self.deferred.reset();
-                                self.deferred.execute( function( process ) {
-                                    if ( options.hasOwnProperty( "workflow" ) ) {
-                                        // this is needed for notifying the workflow editor form
-                                        options.version = version;
-                                        self.model.get( 'postchange' )( process, self );
-                                    } else {
-                                        self._buildModel( process, { id : id, version : version } );
-                                    };
-                                });
+                                self.model.set( 'id', options.id.replace( options.version, this.version ) );
```

Comment:
Ah, didn't relaize that was already here. In any case this isn't necessary anymore with https://github.com/galaxyproject/galaxy/pull/4119, would be good if we can remove that.